app/stac_api/serializers.py

```diff
@@ -521,13 +519,16 @@ def update_or_create(self, look_up, validated_data):
         return asset, created
 
     def validate(self, attrs):
-        if not self.partial:
-            validate_asset_name_with_media_type(attrs.get('name'), attrs.get('media_type'))
-        if self.partial and ('name' in attrs or 'media_type' in attrs):
-            validate_asset_name_with_media_type(
-                attrs.get('name', self.instance.name),
-                attrs.get('media_type', self.instance.media_type)
+        name = attrs['name'] if not self.partial else attrs.get('name', self.instance.name)
+        if self.partial and 'media_type' not in attrs:
+            media_type = validate_and_get_media_type(name, self.instance.media_type)
+            validate_asset_name_with_media_type(name, media_type)
+        else:
+            media_type = validate_and_get_media_type(
+                name, attrs['media_type'], media_type_str_is_normalized=False
             )
+            validate_asset_name_with_media_type(name, media_type)
+            attrs['media_type'] = media_type.media_type_str
```

Comment:
As for the model in the `validate()` method I would try to keep only the field dependencies validations. And try to validate/normalize the media type in a field validator.